docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
-    * For one-time connection this correspondence is not saved in long term storage. And if connected later the same MAC address may get the new integer value in correspondence.
-    * It is not defined what happens if two devices with the same MAC address would connect.
-  * Application ID: is the application unique identifier. It is the incremental integer value assigned and used internally in TM.
-  * Connection ID:
-    * Connection is represented with a unique pair of “Device ID” and “Application ID”.
-    * Connection ID is a system wide unique identifier (incremented integer value) assigned to each new connection.
-    * This ID is used for sending/receiving data and for closing the connection.
-    * There are no certain rules to define how exactly this ID is assigned.
-
-
-### Connection Closing
-
-  * When TM receives ‘close connection’ or ‘disconnect’ commands it tries to finish sending accumulated messages and then closes the connection.
-  * If connection is lost TM will drain all accumulated messages and confirm connection closure.
-
-### SDL Shutting Down
-
-  * When SDL is going to shutdown, TM clears all objects which it has created (e.g. default transport adapters).
-  * Objects created by developer (e.g. developer’s transport adapter or listener) are not removed by TM. The developer must take care of destructing his customized objects by himself.
-
-## Default TM Instance
-
-For creating the instance of the TM with the default configuration it is necessary to use the following code
+Each Transport Adapter is responsible for one specific type of transmitting data, such as TCP or Bluetooth. The Transport Adapter will contain the code to connect and disconnet devices, as well as transfer data. Many of the existing Transport Adapters also implement workers such as a Device Scanner, a Client Connection Listener, or a Server Connection Factory. Currently, Transport Adapters are registered with the Transport Manager within the `TransportManagerDefault::Init()` method; you can add code here to include your custom Transport Adapter. Depending on your implementation, most of the funtionality of the Transport Adapter will likely live in the workers. Two big functions that will for sure need to be implemented in a Transport Adapter are Store() and Restore() which are used to save and resume the state of the Adapter. In the case of the TCP Transport Adapter, the Store() function will save a list of devices' names and addresses, along with the applications each device was running and their corresponding port number. When resuming, Restore() will reconnect to the devices saved in the last state and resumes communication with the the applications on each device. Similar to Transport Managers, other components in Core are able to register a Transport Adapter Listener with a Transport Adapter to later receive events from the Adapter such as `OnDeviceAdded`.
```

Comment:
Each Transport Adapter is responsible for one specific type of *connection*

Reply:
connect and disconnect devices, as well as send and receive data.

Reply:
~Many of the existing Transport Adapters also implement workers such as a Device Scanner, a Client Connection Listener, or a Server Connection Factory.~

Depending on the transport type, a transport adapter may implement subcomponents such as a Device Scanner, a Client Connection Listener, and a Server Connection Factory. 

Make sure those components are explained later too

Reply:
>`TransportManagerDefault::Init()` method;

Would be good to link to that method on github

Reply:
...save and resume the state of the Adapter in case there is an unexpected disconnect or SDL Core is restarted.

Reply:
> Similar to Transport Managers, other components in Core are able to register a Transport Adapter Listener with a Transport Adapter to later receive events from the Adapter such as `OnDeviceAdded`.

I would move this to some place else. Its confusing that it is the last sentence of a large paragraph before a code example that does not have to do with a listener.